### ISSUE TITLE
Feature: Notification if global version is outdated

### ISSUE THREAD

**Issue by yoshuawuyts** [gulpjs/gulp#166]
**tl;dr**: Notifying users when they're using an outdated Gulp installation keeps them updated and prevents unnecessary opening of issues.

It would be nice if we could get a warning when gulp install in `global` is outdated. I know some CLI tools such as Yeoman have implemented it. Ideally users check every day if any of their packages are outdated, but in reality they don't. If you add version detection users will always stay up to date, preventing any confusion when opening issues / opening duplicate issues about fixed topics. 
If such a version detection gets implemented asynchronously I don't believe it won't provide too much of an efficiency cost right?

What do you think of it?
